### PR TITLE
Fix for calc position

### PR DESCRIPTION
**Pull request by miripiruni**
#174 Fixed.

I want to cherry-pick the commit to 4.x and 5.x.

### CODE REVIEW

**Comment by zxqfox** [comment]
Need to check cases with nested arrays, nulls, and single item.

Something like:
```js
[ 'before', { block: 'b1' }, [ { block: 'b1' } ], 'a' ]
[ null, { block: 'b1' }, null, { block: 'b1' }, null ]
[ { block: 'b1' } ]
```

**Comment by miripiruni** [comment]
@zxqfox 
nested arrays: https://github.com/bem/bem-xjst/blob/issue_174/test/bemcontext-position-test.js#L58
nulls: https://github.com/bem/bem-xjst/blob/issue_174/test/bemcontext-position-test.js#L17
single item: https://github.com/bem/bem-xjst/blob/issue_174/test/bemcontext-position-test.js#L79

Or you talk about more complex test?

File: test/bemcontext-position-test.js [hunk 1]
```diff
@@ -92,4 +105,14 @@ describe('BEMContext this.position', function() {
     { block: 'b', content: { elem: 'e' } },
     '<div class="b"><div class="b__e">1</div></div>');
   });
+
+  it('should calc position for nested blocks', function() {
+    test(function() {
+      block('*').cls()(function() {
```
Comment (zxqfox) [inline test/bemcontext-position-test.js]: прямо в класс!?

**Comment by zxqfox** [comment]
Can we add skippage counting for objects without block/elem fields: `{ block: undefined, elem: undefined, text: 'I‘m not a BEM entity' }`?

:ok_hand: at all!

**Comment by miripiruni** [comment]
@tadatuta what do you think? May I merge it?

**Comment by tadatuta** [comment]
I doubt if it's really patch. Potentially it may break someone's project.

**Comment by miripiruni** [comment]
@tadatuta oh, it’s complicated. It’s bug. Shall we treat it as major?

**Comment by tadatuta** [comment]
@miripiruni when it was introduced first?

**Comment by miripiruni** [comment]
Break counting from 2? ;)

**Comment by zxqfox** [comment]
Well, potentially, "someone" who face this problem first should have been reported it. I assume there are no "someone"s since there are no reports from them. Anyway, it's not a behaviour change, it's a fix. It's definitely not a major.

**Comment by tadatuta** [comment]
@zxqfox that someone may think it's just ok. knowledge that we count BEM entities and not any node — is just inside our heads, users don't know it should work this way, they try and see it counts everything.

so there could be templates which use current behaviour as a feature or implement fixes on project level which will fail after this PR.

**Comment by zxqfox** [comment]
No, sorry, but when you have `position = 2` in `{block: 'alone'}` — it's not ok. Definitely.

**Comment by tadatuta** [comment]
well, can't argue with that :)

**Comment by veged** [comment]
LGTM

**Comment by miripiruni** [comment]
Docs needed.

**Pull request merged**